### PR TITLE
Rename Autobackend() `weights` to `model`

### PR DESCRIPTION
**Pull request by glenn-jocher**
@ultralytics/yolo may eliminate duplicate weights in memory

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Refactors AutoBackend to use a consistent model parameter, unifies PyTorch model loading paths, and improves safety/perf by disabling gradients for PyTorch models. 🚀

### 📊 Key Changes
- Replaced weights= with model= across:
  - AutoBackend signature and docstrings
  - Predictor.setup_model() and Validator.__call__() instantiations
- Unified PyTorch handling:
  - Merged in-memory torch.nn.Module and .pt file logic into one path
  - Ensures proper device placement and optional layer fusion (with Jetson-specific ordering)
  - Disables gradients for all PyTorch models by default (p.requires_grad = False)
- Cleanups and fixes:
  - Avoids shadowing the model parameter by not reassigning it to None
  - Clearer warning message when metadata is missing
  - Updated examples to use model="..."

Minimal usage example:
```python
from ultralytics.nn.autobackend import AutoBackend
model = AutoBackend(model="yolo11n.pt", device="cuda")
```

### 🎯 Purpose & Impact
- Consistent API 🧭: Standardizes on model=, reducing confusion between “weights” vs “model” across the codebase.
- More robust PyTorch loading 🧱: Single code path lowers maintenance overhead and edge-case bugs (especially with in-memory models).
- Safer and faster inference ⚡: Gradients are disabled by default for PyTorch models, reducing memory and improving runtime.
- Jetson reliability 🤖: Maintains correct fuse/device order for JetPack 5 devices, improving compatibility.
- Minor migration note 🔁: If you instantiate AutoBackend directly, switch from weights= to model=. Positional use remains unaffected.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response — an Ultralytics engineer will review and assist shortly. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

Additional notes for this PR:
- Please confirm all call sites now use `AutoBackend(model=...)` and that any prior `weights=` keyword usages have been updated across predictors, validators, docs, and examples (e.g., `yolo11n.pt`) 📦
- Since gradients are now disabled for PyTorch inference models, consider adding/expanding tests to validate behavior for both in-memory `torch.nn.Module` models and weight-loaded models ✅
- If there’s any potential backward-compatibility concern for users relying on the `weights` keyword, consider adding a deprecation shim or clear user-facing notes in the docs and release notes 🧭
- Ensure CI covers the new initialization path and that export/inference flows remain unaffected across formats 💡

**Comment by glenn-jocher** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)<sub>

GitHub Actions below triggered via workflow dispatch on this PR branch `autobackend-model` at 2025-08-30 17:57:41 UTC with `@ultralytics/run-ci` command:

* ✅ [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/runs/17346869895): `ci.yml` run 11045
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/17346869925): `docker.yml` run 1016

**Comment by UltralyticsAssistant** [comment]
🚀 Merged and marvelous! Huge thanks, @glenn-jocher, for the thoughtful refactor in [PR #21881](https://github.com/ultralytics/ultralytics/pull/21881). This unification of PyTorch loading paths, the consistent `model=` API, and safe-by-default gradient disabling bring clarity, performance, and reliability—especially for Jetson users.

“Simplicity is the ultimate sophistication.” — Leonardo da Vinci. Your changes embody this: fewer code paths, clearer parameters, and faster inference make YOLO11 workflows smoother for everyone.

Quick win for users:
```python
from ultralytics.nn.autobackend import AutoBackend
model = AutoBackend(model="yolo11n.pt", device="cuda")
```

Appreciate the polish, precision, and momentum this brings to Ultralytics. Onward! 🙌
